### PR TITLE
introduce masked loss feature into the model

### PR DESCRIPTION
**Pull request by Yaoyx**

### CODE REVIEW

File: basenji/dataset.py
Comment (gfudenberg) [inline basenji/dataset.py:1]: this looks good for now, we can discuss down the line if we'd need to have a separate masked_dataset to preserve backwards compatibility.  (though that might not be relevant as David et al are moving to a new repository for model specification)

File: basenji/seqnn.py [hunk 1]
```diff
@@ -26,14 +26,69 @@
...
+        y = y * mask
+        loss = self.compute_loss(x, y, y_pred, sample_weight=None)
+        tf.print(loss)
+        tf.print('loss - unmask_loss:', loss - unmask_loss)
+    self._validate_target_and_loss(y, loss)
+    # Run backwards pass.
+    self.optimizer.minimize(loss, self.trainable_variables, tape=tape)
+    return self.compute_metrics(x, y, y_pred, sample_weight=None)
+
+    # with tf.GradientTape() as tape:
```
Comment (gfudenberg) [inline basenji/seqnn.py]: any reason to comment this out? maybe best to delete?
